symengine/ruby/lib/symengine/basic.rb

```diff
@@ -0,0 +1,13 @@
+require 'set'
+
+module SymEngine
+  class Basic
+    def free_symbols
+      pr_free_symbols.to_set
+    end
+
+    def hash
+      [self.to_s].hash
```

Comment:
Ok. Any ideas for casting `size_t` to `long`? Implementation of `size_t` varies and `#hash` returns a `FIXNUM`, whose size is about the size of `long`, and also there is a macro for converting `long` to `FIXNUM`.